bindings/python/tests/test_parse.py

```diff
@@ -32,3 +32,179 @@ def get_parsed_info(tmp_path, file_contents):
     )
 
     return parsed_info
+
+
+def test_struct_decl(tmp_path):
+    file_contents = "struct AStruct {};"
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+
+    assert struct_decl["kind"] == "STRUCT_DECL"
+    assert struct_decl["name"] == "AStruct"
+
+
+def test_cxx_base_specifier(tmp_path):
+    file_contents = """
+    struct BaseStruct {};
+    struct DerivedStruct: public BaseStruct {};
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    child_struct_decl = parsed_info["members"][1]
+    cxx_base_specifier = child_struct_decl["members"][0]
+
+    assert cxx_base_specifier["kind"] == "CXX_BASE_SPECIFIER"
+    assert cxx_base_specifier["access_specifier"] == "PUBLIC"
+    assert cxx_base_specifier["name"] == "struct BaseStruct"
+
+
+def test_cxx_method(tmp_path):
+    file_contents = """
+    struct AStruct {
+        void aMethod() {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][0]
+
+    assert cxx_method["kind"] == "CXX_METHOD"
+    assert cxx_method["result_type"] == "void"
+    assert cxx_method["name"] == "aMethod"
+
+
+def test_type_ref(tmp_path):
+    file_contents = """
+    struct AStruct { 
+        int aClassMember;
+        void aMethod(AStruct& aClassMember) {}
+    };
+    """
+    parsed_info = get_parsed_info(tmp_path=tmp_path, file_contents=file_contents)
+
+    struct_decl = parsed_info["members"][0]
+    cxx_method = struct_decl["members"][1]
+    l_value_ref = cxx_method["members"][0]
+    type_ref = l_value_ref["members"][0]
+
+    assert type_ref["kind"] == "TYPE_REF"
```

Comment:
Could you also add that `l_value_ref[members][1] == aVariable`?

Reply:
`l_value_ref` (or `parm_decl` as per the updated function) has only one member (index 0).

There is no `aVariable` in the function:
- If you were referring to `aClassMember` (or `aParameter` as per updated), it is the name of the `l_value_ref` (or `parm_decl`) itself.
- If you meant adding a new `aVariable`, it will be another `l_value_ref` (or `parm_decl`).

Reply:
With the new code, I'd ask for a check for `aParameter` by name along with the type